src/presentation/http/router/note.test.ts

```diff
@@ -244,7 +243,128 @@ describe('Note API', () => {
         expect(response?.json().accessRights).toStrictEqual({ canEdit: true });
       });
     });
+  });
+
+  describe('PATCH note/:notePublicId ', () => {
+    test('Update note by public id with 200 status, user is creator of the note', async () => {
+      const expectedStatus = 200;
+      const userId = 2;
+      const accessToken = global.auth(userId);
+
+      const userNote = notes.find(newNote => {
+        return newNote.creator_id === userId;
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${userNote!.public_id}`,
+        body: {
+          'content': { new: 'content added' },
+        },
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+    });
+
+    test('Returns status 401 when the user is not authorized', async () => {
+      const expectedStatus = 401;
+      const correctID = 'Pq1T9vc23Q';
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note/${correctID}`,
+        body: {},
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'You must be authenticated to access this resource' });
+    });
+
+    test('Returns status 406 when the public id does not exist', async () => {
+      const expectedStatus = 406;
+      const nonexistentId = 'ishvm5qH84';
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note/${nonexistentId}`,
+        body: {},
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+    });
+
+    test.each([
+      { id: 'mVz3iHuez',
+        expectedMessage: 'params/notePublicId must NOT have fewer than 10 characters' },
+
+      { id: 'cR8eqF1mFf0',
+        expectedMessage: 'params/notePublicId must NOT have more than 10 characters' },
+
+      { id: '+=*&*5%&&^&-',
+        expectedMessage: '\'/note/+=*&*5%&&^&-\' is not a valid url component' },
+    ])
+    ('Returns 400 when public id of the note settings has incorrect characters and length', async ({ id, expectedMessage }) => {
+      const expectedStatus = 400;
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note/${id}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json().message).toStrictEqual(expectedMessage);
+    });
+
+    test.todo('Update note by public id with 200 status, user is creator of the note');
```

Comment:
there is already such test case
https://github.com/codex-team/notes.api/pull/153/files#diff-790dbfbfc3ab79faef562482e5930737d9b7c352b95e3a0784972d2429b66abfR249